swap_meet/vendor.py

```diff
@@ -1,2 +1,49 @@
+from operator import inv
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item)
+            return item
+
+    def get_by_category(self, category):
+        item_list = []
+        for item in self.inventory:
+            if item.category == category:
+                item_list.append(item) 
+        return item_list
+    
+    def swap_items(self, friend_list, my_item, their_item):
+        if my_item in self.inventory:
+            if their_item in friend_list.inventory:
```

Comment:
Chained conditionals like this can be combined with `and`:
```suggestion
        if my_item in self.inventory and their_item in friend_list.inventory:
```